internal/mtls/caprovider_secret.go

```diff
@@ -118,3 +127,55 @@ func (config *CASecretProvider) CreateRegistrationCertificate(name string) (map[
 	}
 	return res, nil
 }
+
+// SignCSR sign a new CertificateRequest and returns the PEM certificate.
+// This function is going to be used a lot, so using config.latestCA ensure
+// that APIServer is not overloaded with that.
```

Comment:
Wouldn't k8s client caching take care of that?

Reply:
No, I tested and it's not caching this at all, and I do not think that it should TBH.